repository/src/main/java/org/alfresco/repo/action/ActionServiceImpl.java

```diff
@@ -279,7 +279,11 @@ public void init()
         {
             configProperties = applicationContext.getBean("global-properties", Properties.class);
         }
-        actionExecutionValidator = new ActionExecutionValidator(configProperties::getProperty, actionDefinitions::containsKey);
+        actionExecutionValidator = new ActionExecutionValidator(configProperties::getProperty, isActionPublicPredicate());
+    }
+
+    protected Predicate<ActionExecutionContext> isActionPublicPredicate() {

```

Comment:
I've changed the type provided for predicate to allow for ExtendedActionServiceImpl override including overriding only isPublic predicate, so that it could check the context the action is called from and enable Records Management actions only for rules.